File: SAMPLES/CLOJURE/test_runtime.clj
;; test_runtime.clj
;; Test des routines runtime (TODO 24)

;; --- Tests de base : println et str ---
(println "=== Test runtime TODO 24 ===")

;; Entier
(println 42)

;; Chaine
(println "hello world")

;; Booleen
(println true)
(println false)

;; nil
(println nil)

;; --- Concatenation via str ---
(println (str "nombre: " 123))
(println (str "bool: " true))

;; --- Comparaisons d egalite ---
(println (= 1 1))
(println (= 1 2))
(println (= "abc" "abc"))

;; --- Vecteurs ---
(def v [10 20 30])
(println (nth v 0))
(println (nth v 1))
(println (count v))

;; --- Maps ---
(def m {:a 1 :b 2})
(println (get m :a))
(println (get m :b))
(println (count m))

;; --- Ensembles ---
(def s #{1 2 3})
(println (contains? s 2))
(println (count s))

;; --- Arithmetique ---
(println (+ 10 20))
(println (- 100 42))
(println (* 6 7))

(println "=== Fin tests runtime ===")

File: SAMPLES/LUA/test_runtime.lua
-- Test TODO 24 : Runtime minimal
-- Verifie que les routines runtime de base fonctionnent correctement

-- Test 1 : print() affiche une chaine avec retour a la ligne
print("hello")

-- Test 2 : print() affiche un nombre
print(42)

-- Test 3 : print() affiche nil
print(nil)

-- Test 4 : concatenation de chaines
local s = "abc" .. "def"
print(s)

-- Test 5 : longueur de chaine (#)
local n = #"hello"
print(n)

-- Test 6 : tostring() conversion
local t = tostring(123)
print(t)

-- Test 7 : tonumber() conversion
local v = tonumber("456")
print(v)

-- Test 8 : type() retourne le type
local ty = type(10)
print(ty)

-- Test 9 : math.abs() valeur absolue
local a = math.abs(-7)
print(a)

-- Test 10 : comparaison de chaines
local r = ("abc" == "abc")
print(r)

-- Test 11 : table.concat avec separateur
local t2 = {10, 20, 30}
local joined = table.concat(t2, ",")
print(joined)

-- Test 12 : string.sub extraction
local sub = string.sub("hello", 1, 3)
print(sub)
